esmvaltool/diag_scripts/climate_metrics/zec.py

```diff
@@ -0,0 +1,246 @@
+    ProvenanceLogger,
+    get_diagnostic_filename,
+    get_plot_filename,
+    group_metadata,
+    io,
+    run_diagnostic,
+    select_metadata,
+)
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+def _get_default_cfg(cfg):
+    """Get default options for configuration dictionary."""
+    cfg = deepcopy(cfg)
+
+    cfg.setdefault('zec_x', [50])
+    cfg.setdefault(
+        'experiments', {
+            'reference': ['esm-flat10', '1pctCO2'],
+            'simulation': [
+                'esm-flat10-zec', 'esm-1pct-brch-1000PgC',
+                'esm-1pct-brch-750PgC', 'esm-1pct-brch-2000PgC'
+            ]
+        })
+    return cfg
+
+
+def grouped_data(cfg):
+    """Group input data in preparation of ZEC calculation."""
+    input_data = cfg['input_data'].values()
+    group_data = group_metadata(input_data, 'exp')
+    data_grouped = {}
+    for exp in group_data:
+        if exp in cfg['experiments']['simulation']:
+            data_grouped['zecmip_data'] = select_metadata(input_data,
+                                                          short_name='tas',
+                                                          exp=exp)
+        elif exp in cfg['experiments']['reference']:
+            data_grouped['anom'] = select_metadata(input_data,
+                                                   short_name='tas',
+                                                   exp=exp)
+        else:
+            raise ValueError(
+                f"{exp} is not a valid experiment for calculating ZEC, "
+                f"please check the configuration value of 'experiments'. "
+                f"Current accepted experiments are {cfg['experiments']}")
+    if 'zecmip_data' not in data_grouped or 'anom' not in data_grouped:
+        raise ValueError(
+            f"Data does not include experiments valid for ZEC computation, "
+            f"please check the configuration value of 'experiments'. "
+            f"Current accepted experiments are {cfg['experiments']}, "
+            f"received experiments are {list(group_data.keys())}")
+    return data_grouped
+
+
+def calculate_zec(cfg):
+    """Calculate ZEC for each model."""
+    zec = {}
+    data_grouped = grouped_data(cfg)
+    zecmip_data = data_grouped['zecmip_data']
+    anom = data_grouped['anom']
+    for data in zecmip_data:
+        # Account for ensembles by using alias, remove exp name
+        name = data['alias'].replace('_' + data['exp'], '')
+        tas = iris.load_cube(data['filename'])
+        # Match the correct anomaly data, no ensemble key for ensemble mean
+        if '_r' in data['alias']:
+            match_anom = select_metadata(anom,
+                                         dataset=data['dataset'],
+                                         ensemble=data['ensemble'])
+        else:
+            match_anom = select_metadata(anom, dataset=data['dataset'])
+        tas_anom = iris.load_cube(match_anom[0]['filename'])
```

Comment:
```suggestion
        tas_base = iris.load_cube(match_anom[0]['filename'])
```
I think that this is the reference / base tas, against which the anomaly is calculated. If that's correct, then the variable names are confusing. Please use something like `ref` or `base` to denote the baseline, and only use `anom` for the anomaly. 

Or if I've misunderstood, please add a clarifying comment to explain.